icu4c/source/i18n/collationbuilder.cpp

```diff
@@ -307,11 +307,14 @@ CollationBuilder::addReset(int32_t strength, const UnicodeString &str,
             parserErrorReason = "normalizing the reset position";
             return;
         }
-        cesLength = dataBuilder->getCEs(nfdString, ces, 0);
+        cesLength = dataBuilder->getCEs(nfdString, ces, 0, errorCode);
         if(cesLength > Collation::MAX_EXPANSION_LENGTH) {
             errorCode = U_ILLEGAL_ARGUMENT_ERROR;
             parserErrorReason = "reset position maps to too many collation elements (more than 31)";
             return;
+        } else if (errorCode == U_INPUT_TOO_LONG_ERROR) {
```

Comment:
no need to put the if after else here since the previous block will always return and no chance to reach the if check here anyway

Reply:
You're right: there's no need, but I like the "else" there anyway-- I think it's clearer.

But it's not a hill I want to die on, especially after what I've gone through to fix this bug.  I'm changing it.